tt_metal/impl/dispatch/command_queue_interface.hpp

```diff
@@ -4,29 +4,74 @@
 
 #include "tt_metal/common/base.hpp"
 #include "tt_metal/impl/dispatch/device_command.hpp"
+#include "tt_metal/impl/dispatch/dispatch_core_manager.hpp"
 #include "tt_metal/llrt/llrt.hpp"
 #include "tt_metal/common/math.hpp"
 
 using namespace tt::tt_metal;
 
+// Starting L1 address of commands
+inline uint32_t get_command_start_l1_address(bool use_eth_l1) {
+    return (use_eth_l1 ? eth_l1_mem::address_map::ERISC_L1_UNRESERVED_BASE : L1_UNRESERVED_BASE);
+}
+
+// Where issue queue interface core pulls in data (follows command)
+inline uint32_t get_data_section_l1_address(bool use_eth_l1) {
+    uint32_t l1_base = use_eth_l1 ? eth_l1_mem::address_map::ERISC_L1_UNRESERVED_BASE : L1_UNRESERVED_BASE;
+    return l1_base + DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND;
+}
+
+// Space available in issue queue interface core pushing command data to consumer to dispatch or relay forward
+inline uint32_t get_producer_data_buffer_size(bool use_eth_l1) {
+    uint32_t l1_size = use_eth_l1 ? MEM_ETH_SIZE : MEM_L1_SIZE;
+    uint32_t l1_base = use_eth_l1 ? eth_l1_mem::address_map::ERISC_L1_UNRESERVED_BASE : L1_UNRESERVED_BASE;
+    return (l1_size - (DeviceCommand::NUM_ENTRIES_IN_DEVICE_COMMAND * sizeof(uint32_t)) - l1_base);
+}
+
+// Space available in core receiving command data to dispatch or relay forward
+inline uint32_t get_consumer_data_buffer_size(bool use_eth_l1) {
+    uint32_t num_consumer_cmd_slots = use_eth_l1 ? 1 : 2;
+    uint32_t producer_data_buffer_size = get_producer_data_buffer_size(use_eth_l1);
+    return (producer_data_buffer_size - ((num_consumer_cmd_slots - 1) * DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND)) / num_consumer_cmd_slots;
+}
```

Comment:
Should the above functions be templated for easier maintainability?

Reply:
I can use the CoreType rather than bool to handle different cases but generally I think we need a nice way to toggle between different L1 memory maps.